src/main/java/de/tum/in/www1/artemis/service/connectors/athena/AthenaHealthIndicator.java

```diff
@@ -0,0 +1,92 @@
+package de.tum.in.www1.artemis.service.connectors.athena;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import de.tum.in.www1.artemis.service.connectors.ConnectorHealth;
+
+/**
+ * Service determining the health of the Athena service and its assessment modules.
+ */
+@Component
+@Profile("athena")
+public class AthenaHealthIndicator implements HealthIndicator {
+
+    private static final String GREEN_CIRCLE = "\uD83D\uDFE2"; // unicode green circle
+
+    private static final String RED_CIRCLE = "\uD83D\uDD34"; // unicode red circle
+
+    private final RestTemplate shortTimeoutRestTemplate;
+
+    @Value("${artemis.athena.url}")
+    private String athenaUrl;
+
+    public AthenaHealthIndicator(@Qualifier("shortTimeoutAthenaRestTemplate") RestTemplate shortTimeoutRestTemplate) {
+        this.shortTimeoutRestTemplate = shortTimeoutRestTemplate;
+    }
+
+    private record AthenaModuleHealth(String type, boolean healthy, String url) {
```

Comment:
It's about the exercise type (text, programming, modeling, ...), which is relevant because the different modules in Athena are only compatible with one exercise type each. However, I had to think a bit about this myself as well and now decided to rename this variable to `exerciseType` for clarity.